### PR TITLE
Mlml 1814

### PR DESCRIPTION
**Pull request by jonathanbratt**

### CODE REVIEW

File: R/tidy_output.R [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+    token_index = integer(),
+    segment_index = integer(),
+    token = character()
+  )
+  for (i in seq_along(tok_list)) {
+    tokens <- tok_list[[i]]
+    token_ids <- tok_id_list[[i]]
+    segments <- tt_list[[i]]
+    token_seq <- seq_along(tokens)
+    token_map <- dplyr::bind_rows(
```
Comment (jonthegeek) [inline R/tidy_output.R:45]: I'm pretty sure it's generally faster to generate a list of df's in the `for` and then `bind_rows` at the end (the argument to `bind_rows` can be a list of dfs). I'm not going to request that change 'cuz this works and will change quite a bit once we update {tt}, but FYI.

File: R/tidy_output.R [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+#' Takes the raw output from a BERT model and turns the attention weights matrix
+#' into a data frame.
+#'
+#' @inheritParams tidy_bert_output
+#'
+#' @return The attention weights from a BERT model, as a data frame.
+#' @keywords internal
+.process_attention_result <- function(bert_model_output) {
+  attention_output <- bert_model_output$attention_weights
+
```
Comment (jonthegeek) [inline R/tidy_output.R:106]: Should we validate that the output has the shape we expect? Or do that higher up (after adding some classing to {tt} output and then trust it?

File: R/tidy_output.R [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+    # now add columns for attendee tokens
+    dplyr::left_join(token_map,
+                     by = c("sequence_index",
+                            "attention_token_index" = "token_index"),
+                     suffix = c("", "_fill")) %>%
+    dplyr::mutate(attention_segment_index = .data$segment_index_fill,
+                  attention_token = .data$token_fill,
+                  attention_token_id = .data$token_id_fill) %>%
+    dplyr::select(-dplyr::ends_with("_fill")) %>%
+    # I don't think we need to support other conventions for the pad token.
```
Comment (jonthegeek) [inline R/tidy_output.R:185]: I think this is another thing that points at the idea that we should make sure that {tt} tells us that the output we're working with is valid.

**Review by jonthegeek — CHANGES_REQUESTED** [review]
I think I just had one small change for this iteration. But I'm already starting to work on some things to make this way better/safer 🙃

File: R/tidy_output.R (hunk 1 above)
Comment (jonathanbratt) [reply R/tidy_output.R:45]: good call!

File: R/tidy_output.R (hunk 2 above)
Comment (jonathanbratt) [reply R/tidy_output.R:106]: If we're going to add classing to {tt} output at some point, then I think we let that handle the validation for us. In that case, we could implement this as a method. 🤔  For this iteration, I'm more concerned with getting the capability up. Better error messaging can come in future iterations.

**Review by jonthegeek — APPROVED** [review]
Excellent v1! It inspired cleanup that I think will change things a lot, but in a good way!

**Pull request merged**